src/global_constants.h

```diff
@@ -69,6 +69,10 @@ const int maxStarsNum = 1000;
 
 const int maxLocalPlayers = 2;
 
+// was previously changed to 10000; now only used for certain NPC bounds checking.
+// returned to 8000.
+const int FLBlocks = 8000;
```

Comment:
I had to set 10000 not because of NPCs, this would lead the failure on bigger levels with 21+ sections (using the old algorithm). But, I guess, the new algorithm shouldn't limit the range at all.

Reply:
Right. Now it is used to ensure old (possibly buggy behavior) for certain cases where (if I remember right) NPC collision detection was disabled outside of the standard +/-8000 range.

Reply:
That condition I added by me to fix the crash that happens when some flying NPC gets boosted into the speed of lighting, and immediately causes the game to crash, and I added that check. It's not vanilla, It's mine. You can comment it freely.